R/MultilabelStackingWrapper.R

```diff
@@ -0,0 +1,106 @@
+#' @title Use stacking method (stacked generalization) to create a multilabel learner.
+#'
+#' @description
+#' Every learner which is implemented in mlr and which supports binary
+#' classification can be converted to a wrapped stacking multilabel learner.
+#' Stacking trains a binary classifier for each label using predicted label information of all labels (including the target label)
+#' (by cross validation) as additional features.
+#' During prediction these labels need to be predicted by the level-1 learner.
+#'
+#' Models can easily be accessed via \code{\link{getLearnerModel}}.
+#'
+#' @param level1Learner The learner which produces predicted labels at prediction time. Default is \code{learner}, i.e. the same learner as the passed one.  
+#' @param  cv.folds The number of folds for the inner cross validation to predict labels for the augmented feature space. Default is \code{2}.
+#' @template ret_learner
+#' @references
+#' Montanes, E. et al. (2013)
+#' \emph{Dependent binary relevance models for multi-label classification}
+#' Artificial Intelligence Center, University of Oviedo at Gijon, Spain.
+#' @family wrapper
+#' @family multilabel
+#' @export
+#' @examples
+#' d = getTaskData(yeast.task)
+#' # drop some labels so example runs faster
+#' d = d[, c(1:3, 15:117)]
+#' task = makeMultilabelTask(data = d, target = c("label1", "label2", "label3"))
+#' lrn = makeMultilabelStackingWrapper("classif.rpart", level1Learner = "classif.rpart")
+#' lrn = setPredictType(lrn, "prob")
+#' # train, predict and evaluate
+#' mod = train(lrn, task)
+#' pred = predict(mod, task)
+#' p = performance(pred)
+#' performance(pred, measure = hamloss)
+#' getMultilabelBinaryPerformances(pred, measures = list(mmce, auc))
+#' # above works also with predictions from resample!
+makeMultilabelStackingWrapper = function(learner, level1Learner = learner, cv.folds = 2) {
```

Comment:
Should we provide different learner for the two phases? I think rather not (and if yes we should maybe also do it for nested stacking)